File: HomeWorks/HomeWorks013/Core/Models/Circle.py
from Core.Models.Figure import Figure
from math import pi


class Circle(Figure):

    def __init__(self, radius):
        if radius > 0:
            self.radius = radius
        else:
            raise ValueError("Invalid radius")

    def area(self):
        return pi * self.radius ** 2

    def circumference(self):
        return 2 * pi * self.radius

    def perimeter(self):
        return self.circumference()

File: HomeWorks/HomeWorks009/Circle.java
package HomeWorks.HomeWorks009;

public class Circle extends Figure implements Circumference{
    private double radius;

    public Circle(double radius) throws IllegalArgumentException {
        if (radius <= 0) throw new IllegalArgumentException();

        this.radius = radius;
    }

    @Override
    double area() {
        return Math.PI*Math.pow(this.radius, 2);
    }

    @Override
    public double circumference() {
        return 2*Math.PI*this.radius;
    }
    
}

